hpx/runtime/actions/transfer_base_action.hpp

```diff
@@ -19,24 +19,124 @@
 #include <hpx/runtime/serialization/base_object.hpp>
 #include <hpx/runtime/serialization/input_archive.hpp>
 #include <hpx/runtime/serialization/output_archive.hpp>
+#include <hpx/runtime/serialization/unique_ptr.hpp>
 #include <hpx/traits/action_does_termination_detection.hpp>
 #include <hpx/traits/action_message_handler.hpp>
 #include <hpx/traits/action_was_object_migrated.hpp>
 #include <hpx/traits/action_priority.hpp>
 #include <hpx/traits/action_schedule_thread.hpp>
 #include <hpx/traits/action_serialization_filter.hpp>
 #include <hpx/traits/action_stacksize.hpp>
+#include <hpx/util/assert.hpp>
 #include <hpx/util/get_and_reset_value.hpp>
 #include <hpx/util/serialize_exception.hpp>
+#include <hpx/util/tuple.hpp>
 
 #include <boost/atomic.hpp>
 
 #include <cstddef>
 #include <cstdint>
+#include <memory>
+#include <type_traits>
 #include <utility>
 
 namespace hpx { namespace actions
 {
+    ///////////////////////////////////////////////////////////////////////////
+    // If one or more arguments of the action are non-default-constructible,
+    // the transfer_action does not store the argument tuple directly but a
+    // unique_ptr to the tuple instead.
```

Comment:
Also, `tuple` now _can_ be serialized even if not default-constructible, see: https://github.com/STEllAR-GROUP/hpx/blob/fixing_2847/hpx/util/tuple.hpp#L1103-L1121